### ISSUE TITLE
AttributeError: module 'numpy' has no attribute 'int'.

### ISSUE THREAD

**Issue by kkatugam** [DeepLabCut/DeepLabCut#2375]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Attempting to run multi-animal tracking with analyze_videos, keep getting the following error message:

```
Traceback (most recent call last):
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\deeplabcut\gui\utils.py", line 22, in run
    self.func()
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\deeplabcut\pose_estimation_tensorflow\predict_videos.py", line 640, in analyze_videos
    convert_detections2tracklets(
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\deeplabcut\pose_estimation_tensorflow\predict_videos.py", line 1873, in convert_detections2tracklets
    trackingutils.fill_tracklets(
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\deeplabcut\pose_estimation_tensorflow\lib\trackingutils.py", line 761, in fill_tracklets
    tracklet_id, pred_id = content[-2:].astype(np.int)
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\numpy\__init__.py", line 305, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'int'.
```

### Operating System

Windows 11

### DeepLabCut version

dlc version 2.3.5

### DeepLabCut mode

multi animal

### Device type

gpu RTX 3080 laptop

### Steps To Reproduce

1. With a top-down multi-animal project
2. 4 markers per animal per frame, ~15 animals per video
```yaml
individuals:
- bird01
- bird02
- bird03
- bird04
- bird05
- bird06
- bird07
- bird08
- bird09
- bird10
- bird11
- bird12
- bird13
- bird14
- bird15
uniquebodyparts: []
multianimalbodyparts:
- head
- shoulder
- tailbase
- tailend
bodyparts: MULTI!

    # Fraction of video to start/stop when extracting frames for labeling/refinement
start: 0
stop: 1
numframes2pick: 20

    # Plotting configuration
skeleton:
- - head
  - shoulder
- - shoulder
  - tailbase
- - tailbase
  - tailend
skeleton_color: white
pcutoff: 0.6
dotsize: 12
alphavalue: 0.7
colormap: rainbow

    # Training,Evaluation and Analysis configuration
TrainingFraction:
- 0.95
iteration: 0
default_net_type: dlcrnet_ms5
default_augmenter: multi-animal-imgaug
default_track_method: ellipse
snapshotindex: -1
batch_size: 8

    # Cropping Parameters (for analysis and outlier frame detection)
cropping: false
    #if cropping is true for analysis, then set the values here:
x1: 0
x2: 640
y1: 277
y2: 624

    # Refinement configuration (parameters from annotation dataset configuration also relevant in this stage)
corner2move2:
- 50
- 50
move2corner: true
```

3. Run analyze_videos from GUI
4. See attribute error

### Relevant log output

```shell
Traceback (most recent call last):
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\deeplabcut\gui\utils.py", line 22, in run
    self.func()
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\deeplabcut\pose_estimation_tensorflow\predict_videos.py", line 640, in analyze_videos
    convert_detections2tracklets(
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\deeplabcut\pose_estimation_tensorflow\predict_videos.py", line 1873, in convert_detections2tracklets
    trackingutils.fill_tracklets(
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\deeplabcut\pose_estimation_tensorflow\lib\trackingutils.py", line 761, in fill_tracklets
    tracklet_id, pred_id = content[-2:].astype(np.int)
  File "C:\Users\kavya\anaconda3\envs\snakes\lib\site-packages\numpy\__init__.py", line 305, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'int'.
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by andersen1598** [comment]
Similar error:

```python
C:\Users\Rm425\anaconda3\envs\deeplabcut\lib\site-packages\tensorflow\python\keras\engine\base_layer_v1.py:1694: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  warnings.warn('`layer.apply` is deprecated and '
2023-09-13 15:51:42.878471: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-13 15:51:43.210691: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1616] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 9401 MB memory:  -> device: 0, name: NVIDIA GeForce RTX 4080 Laptop GPU, pci bus id: 0000:01:00.0, compute capability: 8.9
2023-09-13 15:51:43.245069: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:354] MLIR V1 optimization pass is not enabled
113it [00:00, 439.69it/s]
0it [00:02, ?it/s]
Traceback (most recent call last):
  File "C:\Users\Rm425\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\gui\utils.py", line 22, in run
    self.func()
  File "C:\Users\Rm425\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\pose_estimation_tensorflow\predict_videos.py", line 640, in analyze_videos
    convert_detections2tracklets(
  File "C:\Users\Rm425\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\pose_estimation_tensorflow\predict_videos.py", line 1873, in convert_detections2tracklets
    trackingutils.fill_tracklets(
  File "C:\Users\Rm425\anaconda3\envs\deeplabcut\lib\site-packages\deeplabcut\pose_estimation_tensorflow\lib\trackingutils.py", line 761, in fill_tracklets
    tracklet_id, pred_id = content[-2:].astype(np.int)
  File "C:\Users\Rm425\anaconda3\envs\deeplabcut\lib\site-packages\numpy\__init__.py", line 305, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'int'.
`np.int` was a deprecated alias for the builtin `int`. To avoid this error in existing code, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

**Comment by MMathisLab** [comment]
If you install `pip install numpy==1.19` does this fix the issue for you?

**Comment by kkatugam** [comment]
> If you install `pip install numpy==1.19` does this fix the issue for you?

I tried to do this install and got the following error

```yaml
pip install numpy==1.19
Collecting numpy==1.19
  Downloading numpy-1.19.0.zip (7.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.3/7.3 MB 17.3 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.


...


      RuntimeError: Running cythonize failed!
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
```

**Comment by kkatugam** [comment]
> If you install `pip install numpy==1.19` does this fix the issue for you?

I was able to bypass the aforementioned error by downgrading to python 3.8, numpy 1.19, and tensorflow 2.7.4 (as well as downgrading many other packages/dependencies). If I'm not mistaken, the DLC documentation states that DLC should work with python versions 3.8, 3.9, or 3.10; and TensorFlow versions 1.0 to 1.15, and 2.0 to 2.10. 

**Do specific features of DLC (i.e., multianimal tracking) only work with specific versions of python/numpy/tensorflow within the ranges mentioned in the documentation?**

**Comment by n-poulsen** [comment]
All of these different versions should work with all features of DeepLabCut. However, installing a specific version of one package will force your hand for other packages. From the [release notes](https://numpy.org/devdocs/release/1.19.0-notes.html) of `numpy 1.19`:
> the Python versions supported for this release are 3.6-3.8

Numpy deprecated `np.int` in version `1.20.0` (see their [release notes](https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated)), and was expired in version `1.24.0` ([release notes: expired deprecations](https://numpy.org/devdocs/release/1.24.0-notes.html#expired-deprecations)). So you can run `pip install "numpy<1.24.0"` and should only get a warning (the code will run).

On our side, the issue `np.int was a deprecated alias for the builtin int` was fixed by #2270. So you can also resolve this `numpy` issue by installing the latest version of DeepLabCut (by cloning the repository and installing it using the conda environments, as described in our [installation docs](https://deeplabcut.github.io/DeepLabCut/docs/installation.html#step-2-build-an-env-using-our-conda-file)).
